File: services/spring/inventory-service/src/main/java/com/smartfulfillment/inventory_service/entity/GlobalInventory.java
package com.smartfulfillment.inventory_service.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "global_inventory")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class GlobalInventory {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "product_id", nullable = false, unique = true)
    private UUID productId; // logical link to product

    @Column(name = "total_stock", nullable = false)
    private int totalStock;

    @Column(name = "reserved_stock", nullable = false)
    private int reservedStock;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: services/go/inventory-service-twin/internal/models/global_inventory.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// GlobalInventory maps to Spring's GlobalInventory entity and `global_inventory` table.
type GlobalInventory struct {
	ID           uuid.UUID `gorm:"type:uuid;primaryKey;column:id" json:"id"`
	ProductID    uuid.UUID `gorm:"type:uuid;not null;unique;column:product_id" json:"productId"`
	TotalStock   int       `gorm:"not null;column:total_stock" json:"totalStock"`
	ReservedStock int      `gorm:"not null;column:reserved_stock" json:"reservedStock"`
	UpdatedAt    time.Time `gorm:"column:updated_at;autoUpdateTime" json:"updatedAt"`
}

func (GlobalInventory) TableName() string {
	return "global_inventory"
}

